docs/conf.rst

```diff
@@ -0,0 +1,253 @@
+.. _sender_conf:
+
+=============
+Configuration
+=============
+
+When constructing a :ref:`sender <sender>` you can pass a configuration string
+to the :func:`questdb.ingress.Sender.from_conf` method.
+
+.. code-block:: python
+
+    from questdb.ingress import Sender
+
+    conf = "http::addr=localhost:9009;username=admin;password=quest;"
+    with Sender.from_conf(conf) as sender:
+        ...
+
+The format of the configuration string is::
+
+    <protocol>::<key>=<value>;<key>=<value>;...;
+
+.. note::
+
+    * The keys are case-sensitive.
+    * The trailing semicolon is mandatory.
+
+The valid protocols are:
+
+* ``tcp``: ILP/TCP
+* ``tcps``: ILP/TCP with TLS
+* ``http``: ILP/HTTP
+* ``https``: ILP/HTTP with TLS
+
+If you're unsure which protocol to use, see :ref:`sender_which_protocol`.
+
+Only the ``addr=host:port`` key is mandatory. It specifies the hostname and port
+of the QuestDB server.
+
+Connection
+==========
+
+* ``addr`` - ``str``: The address of the server in the form of
+  ``host:port``.
+
+  This key-value pair is mandatory, but the port can be defaulted.
+  If omitted, the port will be defaulted to 9009 for TCP(s)
+  and 9000 for HTTP(s).
+
+* ``bind_interface`` - TCP-only, ``str``: Network interface to bind from.
+  Useful if you have an accelerated network interface (e.g. Solarflare) and
+  want to use it.
+  
+  The default is ``0.0.0.0``.
+
+
+Authentication
+==============
+
+HTTP Bearer Token
+-----------------
+* ``token`` - ``str``: Bearer token for HTTP authentication.
+
+HTTP Basic Auth
+---------------
+
+* ``username`` - ``str``: Username for HTTP basic authentication.
+* ``password`` - ``str``: Password for HTTP basic authentication.
+
+TCP Auth
+--------
+
+* ``username`` - ``str``: Username for TCP authentication (A.K.A. *kid*).
+* ``token`` - ``str``: Token for TCP authentication (A.K.A. *d*).
+* ``token_x`` - ``str``: Token X for TCP authentication (A.K.A. *x*).
+* ``token_y`` - ``str``: Token Y for TCP authentication (A.K.A. *y*).
+
+You can additionally set the ``auth_timeout`` parameter (milliseconds) to
+control how long the client will wait for a response from the server during
+the authentication process. The default is 15 seconds.
+
+See the :ref:`auth_and_tls_example` example for more details.
+
+.. _sender_conf_tls:
+
+TLS
+===
+
+TLS in enabled by selecting the ``tcps`` or ``https`` protocol.
+
+See the `QuestDB enterprise TLS documentation <https://questdb.io/docs/operations/tls/>`
+on how to enable this feature in the server.
+
+Open source QuestDB does not offer TLS support out of the box, but you can
```

Comment:
This would make a useful guide.

Let me know if we have any prior art kicking around.

Reply:
All I have is the link that's provided. I have some custom code that I use for testing (see `tls_proxy` in the code base), but that's not production quality.